bluetooth/src/iosMain/kotlin/scanner/Scanner.kt

```diff
@@ -60,53 +68,90 @@ actual class DefaultScanner internal constructor(
             scanner.handleDeviceConnected(didConnectPeripheral.identifier)
         }
 
-        override fun centralManager(central: CBCentralManager, didDisconnectPeripheral: CBPeripheral, error: NSError?) {
+        override fun centralManager(
+            central: CBCentralManager,
+            didDisconnectPeripheral: CBPeripheral,
+            error: NSError?
+        ) {
             scanner.handleDeviceDisconnected(didDisconnectPeripheral.identifier)
         }
 
-        override fun centralManager(central: CBCentralManager, didFailToConnectPeripheral: CBPeripheral, error: NSError?) {
+        override fun centralManager(
+            central: CBCentralManager,
+            didFailToConnectPeripheral: CBPeripheral,
+            error: NSError?
+        ) {
             scanner.handleDeviceDisconnected(didFailToConnectPeripheral.identifier)
         }
     }
 
     private val enabledQueue = dispatch_queue_create("ScannerMonitorEnabled", null)
     private val scanQueue = dispatch_queue_create("ScannerScanning", null)
-    private val pairedDevicesQueue = dispatch_queue_create("ScannerPairedDevices", null)
     override val isSupported: Boolean = true
-    private val centralManagers = sharedMutableListOf<CBCentralManager>()
-    private val discoveringDelegates = sharedMutableListOf<CBCentralManagerDelegateProtocol>()
-    private val activeDelegates = sharedMutableSetOf<CBCentralManagerDelegateProtocol>()
-    override val bluetoothEnabledMonitor: BluetoothMonitor = BluetoothMonitor.Builder { CBCentralManager(null, enabledQueue, emptyMap<Any?, Any>()) }.create()
+    override val bluetoothEnabledMonitor: BluetoothMonitor = BluetoothMonitor.Builder {
+        CBCentralManager(null, enabledQueue, emptyMap<Any?, Any>())
+    }.create()
+
+    private val centralManager = AtomicReference<CBCentralManager?>(null)
+    private val centralManagerDelegate = AtomicReference<CBCentralManagerDelegateProtocol?>(null)
+    private val centralManagerMutex = Mutex()
+    private suspend fun getCentralManager(): CBCentralManager {
+        return centralManager.value ?: centralManagerMutex.withLock {
+            centralManager.value ?: createCentralManager().also {
+                centralManager.set(it)
+            }
+        }
+    }
 
-    private suspend fun scan(filter: UUID? = null) {
-        val centralManager = CBCentralManager(null, scanQueue)
-        centralManagers.add(centralManager)
+    private suspend fun createCentralManager(): CBCentralManager {
+        debug(TAG) { "Create CBCentralManager" }
         val awaitPoweredOn = EmptyCompletableDeferred()
         val delegate = PoweredOnCBCentralManagerDelegate(this, awaitPoweredOn)
-        discoveringDelegates.add(delegate)
-        centralManager.delegate = delegate
-        awaitPoweredOn.await()
+        centralManagerDelegate.set(delegate)
+        val options = mapOf<Any?, Any>(CBCentralManagerOptionShowPowerAlertKey to true)
```

Comment:
Can be moved into constants

Reply:
This also has a side effect of asking the user to enable BT when scanning is started as opposed to when the state machine requests this. If the fix still works without it, please revert this and the change in `generateEnableSensorsActions`